### PR TITLE
Global style refresh

### PR DESCRIPTION
**Pull request by ekfuhrmann**
## Issue
- #7 

## About
@joshuapease Sorry for this being a larger PR than I'd have liked. I originally just wanted to establish colors and whatnot, but in doing so just began updating a bunch of the existing elements.

Overall, this is a style and polish cleanup for the Parts Kit. I narrowed the Tailwind Config styles slightly, reducing the pallet colors amongst some other small tweaks, but still am keeping this relatively vanilla TW.

Lots of atomic styling included in this PR for getting the sidebar and utility bar cleaned up. I've now added `button`, `icon`, and `dropdown` styles for quicker iteration.

With this I also updated the Viewport Size control to be a working dropdown and added the necessary styles/controls to facilitate that. I'm using [Radix](https://www.radix-ui.com/) to handle the component primitive for [the dropdown](https://www.radix-ui.com/primitives/docs/components/dropdown-menu) and will more than likely be looking to Radix again for the Settings modal.

Lastly, I added some improved linting to the project. I added [Tailwind's Prettier Plugin](https://github.com/tailwindlabs/prettier-plugin-tailwindcss) to better manage name sorting, so there may quite a few changes where just those styles are getting moved around. AFAIK these sorting changes don't work for the `js` tailwind files which is a bit of a shame. [There may be a workaround to this limitation but I didn't explore it](https://github.com/tailwindlabs/prettier-plugin-tailwindcss).

## Preview
prev

### CODE REVIEW

File: config/tailwind/dropdown.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const plugin = require('tailwindcss/plugin')
+
+module.exports = plugin(({ addComponents }) => {
+  const dropdown = {
+    '.dropdown': {
+      '@apply inline-flex flex-col gap-2 p-2 bg-white border border-gray-300 shadow min-w-[200px] max-w-[250px] rounded-xl':
```
Comment (joshuapease) [inline config/tailwind/dropdown.js:6]: Shame about the automatic ordering. Do you get intellisene with @apply in these files?

Is the main benefit of creating components JS so you can have intellisense for them in your project?

File: src/features/utility-bar/UtilityBar.tsx [hunk 2]
```diff
@@ -1,49 +1,85 @@
-import SettingsPanel from "../settings/SettingsPanel";
-import { screenSizes, useUtilityBarStore } from "./store";
-import cx from "classnames";
+import cx from 'classnames'
```
Comment (joshuapease) [inline src/features/utility-bar/UtilityBar.tsx:1]: Noticed this wasn't in use

File: src/app.tsx [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
+import UtilityBar from './features/utility-bar/UtilityBar.tsx'
 import {
   findNavItemByUrl,
   findFirstNavItem,
   UseHistory,
   SEARCH_PARAM_PART,
-} from "./features/nav/routing.ts";
-import { uniqueId } from "./utilities/string.ts";
+} from './features/nav/routing.ts'
+import { uniqueId } from './utilities/string.ts'
```
Comment (joshuapease) [inline src/app.tsx:21]: This is the import hell from Solomon's article! 🔥

File: index.html [hunk 4]
```diff
@@ -4,6 +4,14 @@
     <meta charset="UTF-8" />
     <link rel="icon" type="image/svg+xml" href="/vite.svg" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
+    <link rel="preconnect" href="https://fonts.googleapis.com" />
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+    <link
+      href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap"
+      rel="stylesheet"
+    />
```
Comment (joshuapease) [inline index.html:13]: This works great for now... but I don't think these will be bundled with the library when I ship it off to the CDN

**Review by joshuapease — APPROVED** [review]
Ship it!

File: config/tailwind/dropdown.js (hunk 1 above)
Comment (ekfuhrmann) [reply config/tailwind/dropdown.js:6]: So in this case there is no real difference between using CSS or JS. Intellisense should pick up either. The reason I generally keep all of my CSS in JS is for when I do want to create image, I have that availability. And for that reason, I just like to be consistent and keep all styles in JS.

In our current system, as of now, prettier doesn't touch these styles in `@apply`, but looking at the docs for the TW plugin, it seems like there [might be some workarounds](https://github.com/tailwindlabs/prettier-plugin-tailwindcss#sorting-classes-in-template-literals).

File: index.html (hunk 4 above)
Comment (ekfuhrmann) [reply index.html:13]: Yeah, I'll locally host them. Just wanted to make sure the font worked well for our use case before hosting it.

**Pull request merged**